src/marshmallow/schema.py

```diff
@@ -511,19 +514,41 @@ def _serialize(
         .. versionchanged:: 1.0.0
             Renamed from ``marshal``.
         """
-        if many and obj is not None:
-            return [
-                self._serialize(d, many=False)
-                for d in typing.cast(typing.Iterable[_T], obj)
-            ]
-        ret = self.dict_class()
-        for attr_name, field_obj in self.dump_fields.items():
-            value = field_obj.serialize(attr_name, obj, accessor=self.get_attribute)
-            if value is missing:
-                continue
-            key = field_obj.data_key if field_obj.data_key is not None else attr_name
-            ret[key] = value
-        return ret
+        if not self.dump_serializers:
+            accessor = (
+                None
+            )  # type: typing.Optional[typing.Callable[[typing.Any, str, typing.Any], typing.Any]]
+            if self.get_attribute != self.default_get_attribute:
+                accessor = self.get_attribute
+
+            for field_name, field_obj in self.dump_fields.items():
+                key = (
+                    field_obj.data_key if field_obj.data_key is not None else field_name
+                )
+                self.dump_serializers[key] = field_obj.get_serializer(
```

Comment:
It looks like the test failure is being caused by `dump_serializers` being an unordered dict. Currently the ordering is dependent on `self.dump_fields`, which gets initialized to an ordered dict in [schema.py#L1008](https://github.com/marshmallow-code/marshmallow/blob/821c9316be173380f38780da94e26e5e92bf7a70/src/marshmallow/schema.py#L1008).

Adding `self.dump_serializers = self.dict_class()` here should avoid the ordering issue for python3.5.

Reply:
Thanks, I figured as much! I don't have py35/6 on my system, otherwise I would have caught this before pushing :)